File: backend/backend/src/main/java/com/minitransfer/backend/model/Transaction.java
package com.minitransfer.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.index.Indexed;

import java.time.LocalDateTime;

@Document(collection = "transactions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Transaction {

    @Id
    private String id;

    @Indexed
    private String senderId;

    @Indexed
    private String receiverId;

    private Long amount;

    @Builder.Default
    private String status = "SUCCESS";

    @CreatedDate
    private LocalDateTime createdAt;
}

File: mobile/lib/models/transaction.dart
class Transaction {
  final String id;
  final String senderId;
  final String receiverId;
  final int amount;
  final String status;
  final String createdAt;

  Transaction({
    required this.id,
    required this.senderId,
    required this.receiverId,
    required this.amount,
    required this.status,
    required this.createdAt,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['id'],
      senderId: json['senderId'],
      receiverId: json['receiverId'],
      amount: json['amount'],
      status: json['status'],
      createdAt: json['createdAt'],
    );
  }
}